main.tf

```diff
@@ -1,13 +1,93 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled = module.this.enabled
+}
+
+module "iam_role_label" {
+  source  = "cloudposse/label/null"
+  version = "0.24.1"
+  enabled = local.enabled
+
+  attributes = ["role"]
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
-  }
+  context = module.this.context
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+module "iam_policy_label" {
+  source  = "cloudposse/label/null"
+  version = "0.24.1"
+  enabled = local.enabled
+
+  attributes = ["policy"]
+
+  context = module.this.context
+}
+
+resource "aws_transfer_server" "default" {
+  identity_provider_type = "SERVICE_MANAGED"
+  protocols              = ["SFTP"] # SFTP, FTPS, FTP
+  domain                 = "S3"     # EFS, S3
+  endpoint_type          = "PUBLIC" # VPC, PUBLIC
```

Comment:
```
endpoint_type          = var.transfer_endpoint_type
dynamic "endpoint_details" {
    for_each = var.transfer_endpoint_type == "VPC" ? [1] : [0]
    content {
        address_allocation_ids = var.address_allocation_ids
        subnet_ids  = var.subnet_ids
        vpc_id  = var.vpc_id
    }
}
```

Psuedo ish code

Reply:
👍 

Adding this functionality along with adding support for the rest of the protocols on a new PR.